sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/MLIRExporter.cs

```diff
@@ -0,0 +1,853 @@
+        {
+            return semanticModel.GetTypeInfo(method.ReturnType).Type.SpecialType == SpecialType.System_Void;
+        }
+
+        private void ExportBlock(Block block, MethodDeclarationSyntax parentMethod, string functionReturnType)
+        {
+            writer.WriteLine($"^{BlockId(block)}: // {block.GetType().Name}");
+            // MLIR encodes blocks relationships in operations, not in blocks themselves
+            foreach(var op in block.Instructions)
+            {
+                ExtractInstruction(op);
+            }
+            // MLIR encodes blocks relationships in operations, not in blocks themselves
+            // So we need to add the corresponding operations at the end...
+            switch (block)
+            {
+                case JumpBlock jb:
+                    switch (jb.JumpNode)
+                    {
+                        case ReturnStatementSyntax ret:
+                            if (ret.Expression == null)
+                            {
+                                writer.WriteLine($"return {GetLocation(ret)}");
+                            }
+                            else
+                            {
+                                Debug.Assert(functionReturnType!="()","Returning value in function declared with no return type");
+                                var returnedVal = ret.Expression.RemoveParentheses();
+                                if (semanticModel.GetTypeInfo(returnedVal).Type == null &&
+                                    returnedVal.Kind() == SyntaxKind.SimpleMemberAccessExpression)
+                                {
+                                    // Special case a returning a method group that will be cast into a func
+                                    writer.WriteLine($"%{OpId(ret)} = cbde.unknown : none {GetLocation(ret)} // return method group");
+                                    writer.WriteLine($"return %{OpId(ret)} : none {GetLocation(ret)}");
+                                    break;
+                                }
+
+                                var id = ComputeCompatibleId(ret.Expression, functionReturnType);
+                                writer.WriteLine($"return %{id} : {functionReturnType} {GetLocation(ret)}");
+                            }
+                            break;
+                        case BreakStatementSyntax breakStmt:
+                            writer.WriteLine($"br ^{BlockId(jb.SuccessorBlock)} {GetLocation(breakStmt)} // break");
+                            break;
+                        case ContinueStatementSyntax continueStmt:
+                            writer.WriteLine($"br ^{BlockId(jb.SuccessorBlock)} {GetLocation(continueStmt)} // continue");
+                            break;
+                        case ThrowStatementSyntax throwStmt:
+                            // TODO : Should we transfert to a catch block if we are inside a try/catch?
```

Comment:
please open github issue and put link here if there is an action to be done in the future... otherwise what's the purpose of the TODO? https://sonarcloud.io/project/issues?id=sonaranalyzer-dotnet&open=AW8AX2qMPTv0sE1_NOvx&pullRequest=2477&resolved=false